autogluon/task/object_detection/object_detection.py

```diff
@@ -43,15 +43,16 @@ def fit(dataset='voc',
             ngpus_per_trial=1,
             hybridize=True,
             search_strategy='random',
-            search_options={},
+            search_options=None,
             time_limits=None,
             verbose=False,
             transfer='coco',
-            resume='',
+            resume_model='',
```

Comment:
any reason to change the name? This will cause backward incompatibility even though we are in early stages

Reply:
Yes, but it was a bug before. You passed resume (a str) to the scheduler, which needs resume to be a bool. This fixes that bug.

Reply:
The schedulers currently can only resume from the file whose name is passed in checkpoint. This is a little funny, because that same file would then be overwritten by subsequent checkpoints. But it is simple and works, and I'd not change it for now.

Reply:
The defect/bug is unrelated to the name change, we should keep it as resume, to be consistent with image_classification and avoid backwards breaking change.

Reply:
Why not have the boolean resume you require specified in scheduler_options instead?

Reply:
Well, the thing is for all other task's, resume is bool and is passed to the scheduler.

My guess is this code would have never worked, certainly not as intended. In all other task's, you checkpoint the searcher state and resume from there. Here, somehow, you resume from an NN model.

I don't really understand this code. Maybe the best would be to pass resume=False to the scheduler? I can also pass (len(resume) > 0) to the resume (bool) argument of the scheduler, maybe that was intended here? Let me try that.

But I think whoever wrote this task, needs to say what is intended here, because the behaviour differs from the other ones, and that is confusing to the user.

Reply:
OK, I am now passing len(resume) > 0 to resume (bool) of the scheduler. Note what this means: If resume is a non-empty file name, the train_fn will somehow load model pars from that file, and then the scheduler/searcher will resume from the checkpoint loaded from the file name `checkpoint`.

Reply:
@Jerryzcn Could you weigh in here?